### PR TITLE
[PHP] Update CodeIgniter to v4.4

### PR DESCRIPTION
**Pull request by michalsn**
Closes #6658

### CODE REVIEW

File: php/codeigniter4/app/Config/Events.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 use CodeIgniter\Events\Events;
 use CodeIgniter\Exceptions\FrameworkException;
+use CodeIgniter\HotReloader\HotReloader;
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/Events.php:7]: It's active in development mode, but it's there. I don't want to change any default files and configs.

File: php/codeigniter4/app/Config/Modules.php [hunk 2]
```diff
@@ -27,10 +33,33 @@ class Modules extends BaseModules
...
+     *   [
+     *       'exclude' => [
+     *           // List up packages to exclude.
+     *           'pestphp/pest',
+     *       ],
+     *   ]
+     *
+     * @var array
+     */
+    public $composerPackages = [];
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/Modules.php:61]: This is the default config file.

File: php/codeigniter4/app/Config/View.php [hunk 3]
```diff
@@ -14,7 +15,7 @@ class View extends BaseView
      * calls so that it is available to all views. If that is the case,
      * set $saveData to true.
      *
-     * @var boolean
+     * @var bool
      */
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/View.php:19]: This is the default config file.

File: php/codeigniter4/app/Config/Modules.php [hunk 4]
```diff
@@ -27,10 +33,33 @@ class Modules extends BaseModules
...
+     *   ]
+     *   or
+     *   [
+     *       'exclude' => [
+     *           // List up packages to exclude.
+     *           'pestphp/pest',
+     *       ],
+     *   ]
+     *
+     * @var array
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/Modules.php:59]: This is the default config file.

File: php/codeigniter4/app/Config/View.php [hunk 5]
```diff
@@ -14,7 +15,7 @@ class View extends BaseView
...
     public $saveData = true;
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/View.php:20]: This is the default config file.

File: php/codeigniter4/app/Config/View.php [hunk 6]
```diff
@@ -41,4 +42,15 @@ class View extends BaseView
      * @var array
      */
     public $plugins = [];
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/View.php:44]: This is the default config file.

File: php/codeigniter4/app/Config/View.php [hunk 7]
```diff
@@ -41,4 +42,15 @@ class View extends BaseView
...
+
+    /**
+     * View Decorators are class methods that will be run in sequence to
+     * have a chance to alter the generated output just prior to caching
+     * the results.
+     *
+     * All classes must implement CodeIgniter\View\ViewDecoratorInterface
+     *
+     * @var class-string<ViewDecoratorInterface>[]
```
[the remark this answers is not in the document]
Comment (michalsn) [reply php/codeigniter4/app/Config/View.php:53]: This is the default config file.

**Comment by michalsn** [comment]
In general, all these changed config files come from the official release. I can change some things as suggested but it doesn't add much value here.

**Review by waghanza — APPROVED** [review]

**Pull request merged**